rotkehlchen/api/rest.py

```diff
@@ -3906,3 +3908,58 @@ def get_associated_locations(self) -> Response:
             result=_wrap_in_ok_result([str(location) for location in locations]),
             status_code=HTTPStatus.OK,
         )
+
+    def _query_kraken_staking_events(
+        self,
+        only_cache: bool,
+        filter_query: HistoryEventFilterQuery,
+    ) -> Dict[str, Any]:
+        exchanges_list = self.rotkehlchen.exchange_manager.connected_exchanges.get(
+            Location.KRAKEN,
+        )
+        message = ''
+        if exchanges_list is None:
+            events = []
+        else:
+            if only_cache is False:
+                kraken_names = []
+                for kraken_instance in exchanges_list:
+                    with_errors = kraken_instance.query_kraken_ledgers(   # type: ignore
+                        start_ts=filter_query.from_ts,
+                        end_ts=filter_query.to_ts,
+                    )
+                    if with_errors:
+                        kraken_names.append(kraken_instance.name)
+                if len(kraken_names) != 0:
+                    message = f'Failed to query some events from Kraken exchanges {",".join(kraken_names)}'  # noqa: E501
+
+            # After 3865 we should have a recurring tasks that queries for missing prices but
```

Comment:
Don't you essentially implement 3865 at the last commit?

Reply:
Yes! The comment is to let know the reader that after implementing 3865 we have a mechanism to query the prices but we ensure that the retrieve page has prices queried by calling the logic for missing prices

Reply:
yeah so just say that there is a recurring task (notice singular) that queries for missing prices.

"We should have" implies it will happen in the future